pkg/workspace/creds.go

```diff
@@ -30,54 +31,72 @@ import (
 // credentials or tests interacting with one another
 const PulumiCredentialsPathEnvVar = "PULUMI_CREDENTIALS_PATH"
 
-// GetAccessToken returns an access token underneath a given key.
-func GetAccessToken(key string) (string, error) {
+// GetAccount returns an account underneath a given key.
+func GetAccount(key string) (Account, error) {
 	creds, err := GetStoredCredentials()
 	if err != nil && !os.IsNotExist(err) {
-		return "", err
+		return Account{}, err
 	}
-	if creds.AccessTokens == nil {
-		return "", nil
+	if creds.Accounts == nil {
```

Comment:
If I'm reading this correctly, this is the case we'll hit when upgrading to newer versions of Pulumi. The `Accounts` map won't exist in the credentials file, and so we'll look up just the access token from the older `AccessTokens` map. Right?

That seems somewhat problematic since the caller might expect the returned `Account` to have a non-empty `Username` or `LastAccess` value. Right? Is the expectation that an `Account` instance may be nil, but the `AccessToken` will always have a value? Or should callers expect to get an `(Account{}, nil)` response in some cases?

Reply:
> That seems somewhat problematic since the caller might expect the returned Account to have a non-empty Username or LastAccess value. Right?

We are in fact depending on the empty `LastUse` value here in order for the consuming code to work. An empty `LastUse` value will appear to be at the beginning of the Unix epoch, which should be more than an hour ago. This will trigger the validation checks in the backend.

Seems like that validation code should also consider an empty username to mean "needs validation".

Reply:
Having the caller consider an empty `Username` mean "needs validation" seems reasonable. But I would strongly urge you to call that out on the function comment for `GetAccount` to make that explicit. Since otherwise it would very easy easy to assume a `nil` error means the `Account` value is "valid" even if the credential was last used a long time ago.